.github/workflows/cli-verification.yaml

```diff
@@ -0,0 +1,57 @@
+name: CLI Publish & Verify
+
+on:
+    push:
+        branches:
+            - master
+            - staging/**
+    pull_request:
+
+concurrency:
+    group: verify-cli-${{ github.event.pull_request.number || github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    publish-and-test:
+        runs-on: ubuntu-latest
+        env:
+            NANGO_CLI_UPGRADE_MODE: ignore
+        permissions:
+            contents: read
+            packages: write
+        steps:
+            - uses: actions/checkout@v4
+
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: '18.x'
+                  registry-url: 'https://npm.pkg.github.com'
+                  scope: '@nangohq'
+                  always-auth: true
+            - name: Publish npm packages to the github registry
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              shell: bash
+              continue-on-error: true
+              run: |
+                  GIT_HASH=$(git rev-parse HEAD)
+                  bash ./scripts/publish.sh 0.0.1-$GIT_HASH
+            - name: Publish the cli privately under the correct scope
+              working-directory: packages/cli
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              run: |
+                  jq '.name = "@nangohq/cli"' package.json > temp.json && mv temp.json package.json
+                  npm publish --access public
```

Comment:
isn't the cli published by running the `publish.sh` script in the previous step? why do we need to publish it again?

Reply:
It attempts to publish it but fails because it is not scoped under our organization `@nangohq` which is why I rename the package and publish it separately. 

https://github.com/NangoHQ/nango/blob/f9448fcca5d5d6da60630536f2633b4c7a523800/.github/workflows/cli-verification.yaml#L39-L45

Reply:
why not doing that directly then?
Also doesn't it will publish @nangohq/node and @nangohq/shared every time for nothing?

Reply:
> why not doing that directly then?

Easier to reuse the script which bumps the references and packages correctly.

> Also doesn't it will publish @nangohq/node and @nangohq/shared every time for nothing?

Shared needs to be published, node is published again and not used as is frontend but there is no harm in publishing those packages.